tailnet/conn.go

```diff
@@ -290,13 +282,36 @@ func NewConn(options *Options) (conn *Conn, err error) {
 		configMaps:      cfgMaps,
 		nodeUpdater:     nodeUp,
 		telemetrySink:   options.TelemetrySink,
-		telemeteryStore: telemetryStore,
+		telemetryStore:  telemetryStore,
+		createdAt:       time.Now(),
+		watchCtx:        ctx,
+		watchCancel:     ctxCancel,
 	}
 	defer func() {
 		if err != nil {
 			_ = server.Close()
 		}
 	}()
+	if server.telemetryStore != nil {
+		server.wireguardEngine.SetNetInfoCallback(func(ni *tailcfg.NetInfo) {
+			server.telemetryStore.setNetInfo(ni)
+			nodeUp.setNetInfo(ni)
+			if server.telemetryStore.connectedIP != nil {
+				_, _, _, _ = server.Ping(ctx, *server.telemetryStore.connectedIP)
+			}
```

Comment:
We're already checking both the `telemetryStore` and the `connectedIP` aren't nil if that's what you mean. Pinging from the `telemetryStore` would be nicer, but then we'd need it to store a `tailnet.Conn` too, and then they'd both need a pointer to each other and I want to avoid that.

Reply:
You use a mutex when writing to `connectedIP` so you need to use a mutex when reading from it

Reply:
Good point